kitsune/users/migrations/0033_batch_delete_non_migrated_users.py

```diff
@@ -0,0 +1,46 @@
+from django.db import migrations, transaction
+from django.db.models import Q
+
+
+def delete_non_migrated_users(apps, schema_editor):
+    """
+    Delete users where is_fxa_migrated is False and who aren't creators of 
+    any revision, question, or answer, and who don't have any votes or messages
+    """
+    User = apps.get_model('auth', 'User')
+    
+    with transaction.atomic():
+        users_to_delete = User.objects.filter(
+            profile__is_fxa_migrated=False
+        ).exclude(
```

Comment:
You're missing a lot of other related foreign-key relationships here, like `reviewed_revisions` for example. If you don't exclude them as well, you could end up deleting a lot of content via their `on_delete=models.CASCADE`. You'll need to thoroughly review the code for all foreign-key relationships to the user model, and determine which ones need to be added.

Reply:
You just want to make sure that the foreign-key relationships with `on_delete=models.CASCADE` are taken care of, so deleting the user won't trigger the deletion of content that we don't want to be deleted. By "taken care of", I mean either:
- Don't delete the user if the relationship exists.
- Or, replaced the user with something else that makes sense, like `None` or the Sumo Bot or maybe something else that makes more sense.

Reply:
I guess in this case, we're looking for users that we can just cleanly delete without any re-assignment involved, so I guess my second bullet point above doesn't apply.